firmware/lucidgloves-firmware/src/Encoding/IEncoding.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include "../Util/DataStructs.h"
+
+// Interface for encoding
+class IEncoding {
+public:
+    virtual char* encode(int* flexion, int joyX, int joyY, bool joyClick, 
```

Comment:
for now this is probably fine, but I'd imagine the list of parameters might get pretty unwieldy if more inputs are added. Perhaps having a struct that contains all these might be better? This also means you don't have to keep updating the declarations for all the encoding implementations.
Might also want to expand the flexions out of just being an array.